app/src/main/java/com/codesoom/assignment/TaskHttpHandler.java

```diff
@@ -0,0 +1,320 @@
+
+        if (request.isBlank()) {
+            exchange.sendResponseHeaders(400, -1);
+            return;
+        }
+
+        HashMap requestMap = getRequestMap(request);
+        request = (String) requestMap.get("title");
+
+        String content = taskToJson(taskService.createTask(request));
+        exchange.sendResponseHeaders(201, content.getBytes().length);
+        writeResponseBody(exchange, content);
+    }
+
+    /**
+     * 요청된 본문을 HashMap으로 변환해서 리턴합니다.
+     *
+     * @param request 요청된 본문
+     * @return 변환된 HashMap 리턴
+     * @throws JsonProcessingException Json 변환에 문제가 발생할 경우 던집니다.
+     */
+    // 타입을 정의하지 않은 HashMap을 던져도 괜찮을까요?
+    private HashMap getRequestMap(String request) throws JsonProcessingException {
+        return objectMapper.readValue(request, HashMap.class);
+    }
+
+    /**
+     * 응답할 본문을 받아 응답 본문에 담아 보냅니다.
+     *
+     * @param exchange 응답을 담을 본문을 가진 파라미터
+     * @param content 응답 내용
+     * @throws IOException 입출력이 잘못될 경우 던집니다.
+     */
+    private void writeResponseBody(HttpExchange exchange, String content) throws IOException {
+        OutputStream responseBody = exchange.getResponseBody();
+        responseBody.write(content.getBytes());
+        responseBody.flush();
+        responseBody.close();
+    }
+
+    /**
+     * Task의 숫자 형식의 id가 경로에 포함되어 있으면 true, 아니면 false를 리턴합니다.
+     *
+     * @param path 수신된 Http 요청의 경로
+     * @return 숫자 형식의 id가 경로에 포함되어 있으면 true, 아니면 false
+     */
+    private boolean isDetailMatches(String path) {
+        return path.matches("/tasks/[0-9]+");
+    }
+
+    /**
+     * 특정 Task를 GET 요청이 왔을 때, 요청한 Task를 리턴한다.
+     *
+     * @param path 수신된 Http 요청의 경로
+     * @return 요청한 Task를 String으로 변환해서 리턴한다.
+     * @throws JsonProcessingException Json으로 변환하면서 에러가 발생한 경우 던집니다.
+     * @throws NoSuchElementException 요청된 id를 찾지 못했을 경우 던집니다.
+     */
+    private String handleDetailGet(String path) throws JsonProcessingException, NoSuchElementException {
+        String[] splitedPath = path.split("/");
+        Long findId = Long.parseLong(splitedPath[2]);
+
+        Task task = tasks.stream()
+                    .filter(t -> t.getId().equals(findId))
+                    .findFirst()
+                    .orElseThrow();
+
+        return taskToJson(task);
+    }
+
+    /**
+     * DELETE 요청이 왔을 때 경로의 id를 찾아 제거하고 빈 문자열을 리턴한다.
+     *
+     * @param path 수신된 Http 요청의 경로
+     * @return 빈 문자열을 리턴
+     * @throws NoSuchElementException 요청된 id를 찾지 못했을 경우 던집니다.
+     */
+    private String handleDelete(String path) {
+        String[] splitedPath = path.split("/");
+        Long findId = Long.valueOf(splitedPath[2]);
+
+        int index = -1;
+        for (int i = 0; i < tasks.size(); i++) {
+            if (tasks.get(i).getId().equals(findId)) {
+                index = i;
+                break;
+            }
+        }
+
+        if (index == -1) {
+            throw new NoSuchElementException();
+        }
+        tasks.remove(index);
+        return "";
+    }
```

Comment:
```suggestion
    private String handleDelete(String path) {
        Long findId = Long.valueOf(path.split("/")[2]);

        for (int i = 0; i < tasks.size(); i++) {
            if (tasks.get(i).getId().equals(findId)) {
                tasks.remove(i);
                return i;
            }
        }
        throw new NoSuchElementException();
    }
```
이렇게 바꿀 수는 없을까요? 한번 확인부탁드려요!
splitedPath 를 제거한 이유는 코드 자체에서 `split` 하는 것을 알 수 있기 때문이에요!

Reply:
코드 자체에서 `split`하는 것을 알면 어떤 문제가 있나요?

Reply:
뭔가 제가 글을 잘못 쓴 것 같군요! split을 분리해보자는 생각을 했던 것 같은데요 ㅠㅜㅠㅜ

일단 splitedPath를 제거한 이유는 한곳에서만 쓰기 때문입니다.

제가 원하는 코드는 `Long.valueOf(path.split("/")[2]);` 이 코드를 제거하는 것입니다.
`split` 을 알면 문제된다는 것은 `path` 에서 `id` 를 꺼내는 방식을 handleDelete 가 알아버린다는 것입니다!

Reply:
뭔가 제가 글을 잘못 쓴 것 같군요! split을 분리해보자는 생각을 했던 것 같은데요 ㅠㅜㅠㅜ

일단 splitedPath를 제거한 이유는 한곳에서만 쓰기 때문입니다.

제가 원하는 코드는 Long.valueOf(path.split("/")[2]); 이 코드를 제거하는 것입니다.
split 을 알면 문제된다는 것은 path 에서 id 를 꺼내는 방식을 handleDelete 가 알아버린다는 것입니다!